splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java

```diff
@@ -1645,4 +1645,47 @@ public void addCatalogVersion(TransactionController tc) throws StandardException
 
         }
     }
+
+    public void addMinRetentionPeriodColumn(TransactionController tc) throws StandardException {
+        SchemaDescriptor sd = getSystemSchemaDescriptor();
+        TableDescriptor td = getTableDescriptor(SYSTABLESRowFactory.TABLENAME_STRING, sd, tc);
+        ColumnDescriptor cd = td.getColumnDescriptor(SYSTABLESRowFactory.MIN_RETENTION_PERIOD);
+        if (cd == null) { // needs updating
+            tc.elevate("dictionary");
+            dropTableDescriptor(td, sd, tc);
+            td.setColumnSequence(td.getColumnSequence() + 1);
+            // add the table descriptor with new name
+            addDescriptor(td, sd, DataDictionary.SYSTABLES_CATALOG_NUM, false, tc, false);
+
+            ColumnDescriptor columnDescriptor;
+            UUID uuid = getUUIDFactory().createUUID();
+
+            // Add the column MIN_RETENTION_PERIOD
+            DataValueDescriptor storableDV = getDataValueFactory().getNullLong(null);
+            int colNumber = td.getNumberOfColumns() + 1;
+            DataTypeDescriptor dtd = DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.BIGINT, 1);
+            tc.addColumnToConglomerate(td.getHeapConglomerateId(), colNumber, storableDV, dtd.getCollationType());
+
+            columnDescriptor = new ColumnDescriptor(SYSTABLESRowFactory.MIN_RETENTION_PERIOD, colNumber,
+                    colNumber, dtd, null, null, td, uuid, 0, 0, td.getColumnSequence());
+
+            addDescriptor(columnDescriptor, td, DataDictionary.SYSCOLUMNS_CATALOG_NUM, false, tc, false);
+
+            // now add the column to the table's column descriptor list.
+            td.getColumnDescriptorList().add(columnDescriptor);
+            updateSYSCOLPERMSforAddColumnToUserTable(td.getUUID(), tc);
+
+            SpliceLogUtils.info(LOG, String.format("%s upgraded: added a column: %s.", SYSTABLESRowFactory.TABLENAME_STRING,
+                    SYSTABLESRowFactory.MIN_RETENTION_PERIOD));
+
+            // finally, update the SYSTABLESVIEW
+            ViewDescriptor vd=getViewDescriptor(td);
+            dropViewDescriptor(vd, tc);
+            DataDescriptorGenerator ddg=getDataDescriptorGenerator();
+            vd=ddg.newViewDescriptor(td.getUUID(),SYSTABLESRowFactory.SYSTABLE_VIEW_NAME, SYSTABLESRowFactory.SYSTABLE_VIEW_SQL,0,sd.getUUID());
+            addDescriptor(vd,sd,DataDictionary.SYSVIEWS_CATALOG_NUM,true, tc,false);
```

Comment:
Also since the new column you add to the view is not the last column in the view, it is better to drop all the columns in the view together with the view and re-create the view, similar to what we do in SpliceDataDictionary.updateColumnViewInSys().

Reply:
done.